enhancements/ingress/ingress-dashboard.md

```diff
@@ -0,0 +1,199 @@
+- Number/IP/names of backends per route
+- Balancing option/algorithm used by a route
+- Traffic volume (Mbps) on each ingress instance
+- Refresh rate of HAproxy
+- Route's backend failure count
+
+And also global statistics such as:
+- Number of routes / shards / routes per shard
+
+### Workflow Description
+
+A cluster administrator will be able to view this dashboard from the OpenShift Console, in the _Administrator_ view, under _Observe_ > _Dashboards_ menu.
+Several dashboards are already listed there, e.g:
+
+- `Kubernetes / Compute Resources / Pod` (tag: `kubernetes-mixin`)
+- `Kubernetes / Compute Resources / Workload` (tag: `kubernetes-mixin`)
+- `Kubernetes / Networking / Cluster` (tag: `kubernetes-mixin`)
+- `Node Exporter / USE Method / Cluster` (tag: `node-exporter-mixin`)
+- `Node Exporter / USE Method / Node` (tag: `node-exporter-mixin`)
+
+The new ingress dashboard will be listed there, as:
+- `Networking / Ingress` (tag: `networking-mixin`)
+
+This "Networking" category can potentially be used for other Network-related dashboards, such as for OVN, NetObserv, etc.
+
+Clicking on this dashboard will open it, showing time-series charts such as cluster ingress stats and HAProxy metrics.
+On each chart, an "Inspect" link allows to view that metrics from the _Metrics_ page, which allows to customize the query (e.g: modifying the label filters, the grouping, etc.) and view the result directly.
+Note that editing a query from the _Metrics_ page does not affect dashboards displayed in the _Dashboards_ page.
+These behaviours are already implemented in the Console and do not necessitate any change.
+
+### API Extensions
+
+No planned change on the API.
+
+### Implementation Details / Notes / Constraints
+
+The new `ConfigMap` installed in `openshift-config-managed` should be named `grafana-dashboard-ingress` (The `grafana-dashboard-` prefix is common for all such dashboards).
+
+It needs to be labelled with `console.openshift.io/dashboard: "true"`.
+
+Dashboards have a `title` field as part of their Grafana JSON model, which is displayed in the OpenShift Console where dashboards are listed.
+Here `title` should be `Networking / Ingress`.
+
+Dashboards should also have a tag that identifies their supplier in the OpenShift Console; existing tags are: `kubernetes-mixin`, `node-exporter-mixin`, `prometheus-mixin`, `etcd-mixin`.
+A new tag named `networking-mixin` should be used for this new dashboard. This tag aims to group all dashboards related to networking, such as also OVN dashboards and NetObserv.
+This tag is directly set in the static JSON definition of the dashboard. No more action is required for tag creation.
+
+A typical procedure to design and create the dashboard is to use Grafana for designing purpose, then export the dashboard as JSON and save it as an asset in the target repository. Then, it can be embedded in the built artifact using `go:embed`, and injected into a `ConfigMap`. Example:
```

Comment:
You should consider the fact that there can be more than one Ingress controller.  I don't think we want to have one dashboard per Ingress controller, but we need to be able to identify a new, distinct controller that manages the installation and update of a single dashboard.  And that dashboard may need to be able to isolate different ingress controllers' metrics.

Reply:
This is a good point, I made the wrong assumption that IngressController resource was a singleton, now realizing that it isn't.
So, what are our options? we briefly discussed that with @frobware and @OlivierCazade this morning:

- The operator could provide the dashboard `ConfigMap` as a static YAML, as part of its manifests, just like it also provides a [ServiceMonitor](https://github.com/openshift/cluster-ingress-operator/blob/master/manifests/0000_90_ingress-operator_02_servicemonitor.yaml). That's probably the simplest way; the dashboard would be tied to the operator own lifecycle, which I think is what we want at least at the moment. Problem: it could turn out not to be flexible enough for future evolutions. For instance, like I mentioned in this EP, if in the future you want to select dashboards more dynamically based on some IngressController config (I don't know what it could be, but I can imagine using a different proxy than haproxy, which would provide a different set of metrics), then a statically deployed dashboard might be too limited.

- Another option could be to still tie it with ingress controllers, deployed programmatically, but just caring about not deploying a dashboard if one already exists, and not deleting it on ingresscontroller deletion if more ingresscontrollers are still out there. The programmatical approach offers more flexibility, in case the dashboard needs at some point to be modified dynamically, e.g. for tying charts with the metrics presence.

Do you see any other resource managed by the IngressOperator that follows a similar pattern, deployed programmatically, but not tied to a specific IngressController (or to any non-singleton controller)? If there's already a pattern in code to do that, we should probably take inspiration from it.

Reply:
I've added a paragraph below to explain that this is a new controller / what is watches / etc.